### ISSUE TITLE
re2c should install unicode_categories.re somewhere

### ISSUE THREAD

**Issue by terpstra** [skvadrik/re2c#253]
Firstly, re2c v1.2 is great!

I am now using more than half of the new feature included in that release:
- the new EOF rule made it easy to be 8-bit clean (before I terminated on nulls)
- using include files let me shared unicode definitions between files
- i ditched all my \uxxxx escapes and now use input-encoding utf8
- the new difference operator on utf8 charsets saved me several very large classes
- i eliminated my 're2c; mv' work-arounds for 're2c -o'

My only complaint: because re2c does not install 'unicode_categories.re', I had to copy that file into my own source repository. I would have preferred to just include re2c's installed copy in my build flow.

**Comment by skvadrik** [comment]
@terpstra , you should really thank yourself for asking for all those features. :D  

I'm a bit uncertain whether re2c should install `unicode_categories.re` in the usual include path like `/usr/include`. If yes, then it is only a matter of adding the file to `include_HEADERS` Automake variable, as [described here](https://www.gnu.org/software/automake/manual/html_node/Headers.html).

**Comment by terpstra** [comment]
If you're asking my opinion, then I think you should put it into /usr/share/re2c/stdlib.

From the manual of automake:
```
pkgdatadir
The directory for installing idiosyncratic read-only architecture-independent data files for this program. This is usually the same place as ‘datarootdir’, but we use the two separate variables so that you can move these program-specific files without altering the location for Info files, man pages, etc.

The default is $(datarootdir).

The definition of ‘datadir’ is the same for all packages, so you should install your data in a subdirectory thereof. Most packages install their data under $(datadir)/package-name/.
```

**Comment by terpstra** [comment]
To do this, you need:
```
 pkgstdlibdir = $(pkgdatadir)/re2c/stdlib
 dist_pkgstdlib_DATA = $(srcdir)/include/*.re
```

**Comment by terpstra** [comment]
As a bonus feature, you could/should automatically search this 'stdlib' directory when processing re2c include directives. Otherwise, you would need a mechanism (like pkg-config) to inform the user about where re2c was installed (/usr, /usr/local, /opt/local, ...) in order to set the correct '-I' when invoking re2c.

**Comment by skvadrik** [comment]
@terpstra I did as you suggested (https://github.com/skvadrik/re2c/commit/c3880fc84e55a3797b98b823505f02b2006a4006), except that I used `$(datadir)/re2c/stdlib` instead of `$(pkgdatadir)/re2c/stdlib` --- the latter implicitly adds `/re2c`. Special thanks for reminding me to use this location when looking for include files.

**Comment by skvadrik** [comment]
Released re2c-1.2.1 with the fix. Closing the bug for now.
